lib/bandit/http2/stream_task.ex

```diff
@@ -91,9 +91,9 @@ defmodule Bandit.HTTP2.StreamTask do
   defp combine_cookie_crumbs(headers) do
     {crumbs, other_headers} = headers |> Enum.split_with(fn {header, _} -> header == "cookie" end)
 
-    combined_cookie = crumbs |> Enum.map(fn {"cookie", crumb} -> crumb end) |> Enum.join("; ")
+    combined_cookie = Enum.map_join(crumbs, "; ", fn {"cookie", crumb} -> crumb end)
 
-    other_headers ++ [{"cookie", combined_cookie}]
+    [{"cookie", combined_cookie} | other_headers]
```

Comment:
I can't quite recall what the reason was, but I'm pretty sure the appending here is for a specific clause in the RFC. In any case, I certainly wouldn't have written it this way unless there was a definite reason (my personal style is exactly what your suggestion here is). OK to walk this line back?

Reply:
Yeah, no problem. 

Do you remember if it was a http2 specific thing with the header ordering? 
I only found [this](https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.5) mentioning cookies in the RFC.